internal/relay/relay.go

```diff
@@ -115,7 +115,7 @@ func (re *Relay) Start(globalCtx context.Context) error {
 	go func() {
 		defer wg.Done()
 		// Wait till main ctx is cancelled.
-		<-globalCtx.Done()
```

Comment:
globalCtx is only cancelled on SIGINT/SIGTERM. For `stop-at-end` scenarios, we need to force cancel the forked context and gracefully exit all routines. If not, healthcheck goroutines will be lurk around and program hangs waiting for SIGINT/SIGTERM.